src/Gaufrette/Adapter/Ftp.php

```diff
@@ -366,12 +378,31 @@ private function isDir($directory)
             return true;
         }
 
-        if (!@ftp_chdir($this->getConnection(), $directory)) {
-            return false;
+        $chDirResult = false;
+        try {
+            $chDirResult = ftp_chdir($this->getConnection(), $this->directory);
+
+            // change directory again to return in the base directory
+            ftp_chdir($this->getConnection(), $this->directory);
+        } catch(\Exception $e) {
+
+            // Build the FTP URL that will be used to check if the path is a directory or not
+            $url = $this->createConnectionUrl();
+
+            // is_dir is only available in passive mode.
+            // See https://php.net/manual/en/wrappers.ftp.php for more details.
+            if(!$this->passive) {
+                throw new \BadFunctionCallException('is_dir can only be used in passive mode.');
+            }
+
+            if (!@is_dir($url . $directory)) {
+                return false;
+            }
         }
 
-        // change directory again to return in the base directory
-        ftp_chdir($this->getConnection(), $this->directory);
+        if (!$chDirResult) {
```

Comment:
In the end this condition can be avoided (as the final `return true`).
You can directly `return $chDirResult;` after the call to `ftp_chdir` to get back to the original dir ;)

Reply:
Gosh, true. I didn't see this anymore.